frontend/public/components/import-yaml-results.tsx

```diff
@@ -0,0 +1,299 @@
+import { Location } from 'history';
+import { useTranslation } from 'react-i18next';
+import * as _ from 'lodash';
+import { Map as ImmutableMap } from 'immutable';
+import { Bullseye, Button, Spinner } from '@patternfly/react-core';
+import {
+  GreenCheckCircleIcon,
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared';
+import {
+  TableComposable,
+  TableGridBreakpoint,
+  Tbody,
+  Td,
+  Th,
+  Thead,
+  Tr,
+} from '@patternfly/react-table';
+
+import { history, ResourceLink } from './utils';
+import { createRequests } from './edit-yaml';
+import { K8sKind, K8sResourceKindReference } from '../module/k8s';
+import { formatNamespacedRouteForResource } from '@console/shared/src/utils';
+
+const reactPropFix = {
+  translate: 'no',
+};
+
+const ImportYAMLPageStatus: React.FC<ImportYAMLStatusProps> = ({ errors, sending }) => {
+  const { t } = useTranslation();
+  let StatusBlock;
+
+  if (sending) {
+    StatusBlock = (
+      <>
+        <Spinner size="lg" />
+        <h2>{t('public~Creating resources...')}</h2>
+      </>
+    );
+  } else if (!sending && !errors) {
+    StatusBlock = (
+      <>
+        <GreenCheckCircleIcon size="lg" />
+        <h2>{t('public~Resources successfully created')}</h2>
+      </>
+    );
+  } else {
+    StatusBlock = (
+      <>
+        <YellowExclamationTriangleIcon size="lg" />
+        <h2>{t('public~One or more resources failed to create')}</h2>
+      </>
+    );
+  }
+  return <div className="co-import-yaml-status text-center">{StatusBlock}</div>;
+};
+
+const ImportYAMLResourceStatus: React.FC<ImportYAMLResourceStatusProps> = ({ status }) => {
+  let StatusIcon;
+  if (status.creating) {
+    StatusIcon = <Spinner size="sm" className="co-icon-space-r" />;
+  } else if (status.error) {
+    StatusIcon = <RedExclamationCircleIcon className="co-icon-space-r" size="sm" />;
+  } else {
+    StatusIcon = <GreenCheckCircleIcon className="co-icon-space-r" size="sm" />;
+  }
+  return (
+    <span>
+      {StatusIcon}
+      {status.message}
+    </span>
+  );
+};
+
+const IMPORT_COLS = {
+  NAME: 0,
+  NAMESPACE: 1,
+  STATUS: 2,
+};
+const COLUMN_MAP = ['metadata.name', 'metadata.namespace', 'message'];
+
+const getTableCell = (row, column) => {
+  return column === IMPORT_COLS.STATUS
+    ? _.get(row.status, COLUMN_MAP[column])
+    : _.get(row.obj, COLUMN_MAP[column]);
+};
+
+const ImportYAMLResults: React.FC<ImportYAMLResultsPageProps> = ({
+  activeNamespace,
+  location,
+  models,
+}) => {
+  const { t } = useTranslation();
+  const [rows, setRows] = React.useState(
+    location.state.objs.map((obj) => ({
+      obj: { ...obj },
+      status: { creating: true, message: t('public~Creating') },
+    })),
+  );
+  const [sending, setSending] = React.useState(true);
+  const [errors, setErrors] = React.useState(false);
+  const [activeSortIndex, setActiveSortIndex] = React.useState(-1);
+  const [activeSortDirection, setActiveSortDirection] = React.useState(null);
```

Comment:
Since we're initializing state with `null`, TypeScript won't be able to infer a type. Let's declare a type here.